sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -510,6 +510,10 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
         } else {
           e.copy(branches = branches.take(i).map(branch => (branch._1, elseValue)))
         }
+
+      case EqualTo(CaseWhen(branches, _), right)
+          if branches.count(_._2.semanticEquals(right)) == 1 =>
```

Comment:
if there are more than one matches, shall we combine the conditions with `Or`?

Reply:
Yes.